File: NaproKarta.Server/Interfaces/ICycle.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using NaproKarta.Server.Models;

namespace NaproKarta.Server.Interfaces
{
   public interface ICycle
   {
      [Key]
      int Id { get; set; }
      int ChartId { get; set; }
      IChart Chart { get; set; }
      string NumberInChart { get; set; }
      string Note { get; set; }
      IList<Observation> Observations { get; set; }
   }
}

File: NaproKarta.Client/App/src/app/models/icycle.ts
import { IObservation } from './iobservation';

export interface ICycle {
  id: number;
  note: string;
  observations: IObservation[];
  numberInChart: number;
  chartId: number;
}

export class Cycle implements ICycle {
  id: number;
  note: string;
  observations: IObservation[];
  numberInChart: number;
  chartId: number;
}
